src/main/java/io/airlift/units/Duration.java

```diff
@@ -130,6 +135,32 @@ public Duration convertToMostSuccinctTimeUnit()
         return convertTo(unitToUse);
     }
 
+    public java.time.Duration toJavaTime()
+    {
+        long seconds;
+        long nanoAdjustment;
+        long secondsPerUnit = SECONDS.convert(1, unit);
+        long nanosPerUnit = NANOSECONDS.convert(1, unit);
+        if (secondsPerUnit > 1) {
+            //noinspection NumericCastThatLosesPrecision
```

Comment:
Use Math.toIntExact instead of these suppressions.

Reply:
i realized this was due to `Numeric cast that loses precision` inspection flagging double->long cast.
while i think this isn't a bad inspection, i should not impose my non-default setup on everyone